### DISCUSSION TITLE
Is there really no way to configure the Tor connection?

### DISCUSSION THREAD

**Discussion by LEOcab** [bisq-network/bisq#7271] [General]
I cannot figure out how to configure my Tor connection if I am already inside of a Tor-isolated environment, eg. Whonix or Tails. There's [this Github issue](https://github.com/bisq-network/bisq/issues/1761) from 2018 that apparently never got solved. This seems too important to have been overlooked for six years, so I hope I'm wrong.

The Tor developers have stated that running Tor over Tor is not a good idea. Neither the Whonix or Tails instructions for installing Bisq seem to take any extra steps to fix this.

If there really is no way to use Tor other than the built-in package, then can I at least configure a SOCKS proxy somehow? That could be a viable workaround for me, but, I can't even access any settings before connecting, except Tor settings after it fails to connect.

One more thing. Is this the best place to talk about Bisq? It seems the Github issues has been heavily restricted since 2018, and no longer allows such questions. If I can get some confirmation that `bisq.chat` is more popular then maybe I'll consider downloading an app, which is apparently required to enter the chat. (FML if I have to spin up yet another virtual machine JUST to talk to the Bisq people FFS)

**Comment by dutu** [comment]
Yes, you can use existing tor, rather than Bisq's tor, by `--torControlPort` option when launching Bisq.
See [Running Bisq on Tails](https://bisq.wiki/Running_Bisq_on_Tails), or for easy setup [Run-On-Tails: Bisq](https://dutu.github.io/run-on-tails/bisq/bisq.html). 

There is some instruction on how to run Bisq on Whonix (very similar with Tails, but tor control port and onion-grater file would differ a bit).

**Reply by dutu** [reply]
[Bisq can now use an external Tor service](https://github.com/bisq-network/bisq/pull/1935)
